src/lv_objx/lv_canvas.c

```diff
@@ -658,11 +658,36 @@ void lv_canvas_draw_line(lv_obj_t * canvas, const lv_point_t * points, uint32_t
     lv_disp_t * refr_ori = lv_refr_get_disp_refreshing();
     lv_refr_set_disp_refreshing(&disp);
 
+    lv_style_t circle_style_tmp; /*If rounded...*/
+    lv_style_copy(&circle_style_tmp, style);
+    circle_style_tmp.body.radius     = LV_RADIUS_CIRCLE;
+    circle_style_tmp.body.main_color = style->line.color;
+    circle_style_tmp.body.grad_color = style->line.color;
+    circle_style_tmp.body.opa        = style->line.opa;
+    lv_area_t circle_area;
+
     uint32_t i;
     for(i = 0; i < point_cnt - 1; i++) {
         lv_draw_line(&points[i], &points[i + 1], &mask, style, LV_OPA_COVER);
+        
+        /*Draw circle on the joints if enabled*/
+        if(style->line.rounded) {
```

Comment:
It could go before line 661 to not evaluate any round related code.

Reply:
I found some code from lv_line,

https://github.com/littlevgl/lvgl/blob/master/src/lv_objx/lv_line.c#L227-L267

And adjusted for the canvas' api.
If anything is more better, whatever as you choose. ^^

Reply:
Oh, you are right, I didn't spot the issue in my own code. :slightly_smiling_face: 

Do you have time/interest to update both places to use a larger `if`?

Reply:
Sorry for my English is very limit, I don't know the meaning of this wording   

>>> ...  to use a larger if?  <<<

Reply:
I think, as long as the variables are not referenced, they will not actually be emitted (if optimization is enabled). So wrapping [these lines](https://github.com/littlevgl/lvgl/blob/80448a1d541d8e02e6de0415a6f3221fb1f51f13/src/lv_objx/lv_canvas.c#L662-L666) should hopefully be enough. Otherwise, it would be rather messy to duplicate the loop on both halves of the if statement.